src/load/azext_load/data_plane/load_notifications/utils.py

```diff
@@ -0,0 +1,139 @@
+    TestsNotificationRule,
+    TestRunEndedNotificationEventFilter,
+    TestRunStartedNotificationEventFilter,
+    TriggerCompletedNotificationEventFilter,
+    TriggerDisabledNotificationEventFilter,
+    TestRunEndedEventCondition,
+)
+from azure.cli.core.azclierror import InvalidArgumentValueError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+def get_notification_rule_create_body(
+    action_groups=None,
+    event=None,
+    display_name=None,
+    test_ids=None,
+    all_tests=False,
+    all_events=False,
+):
+    """
+    This function creates a NotificationRule object with all the required fields for create request.
+    """
+    logger.debug("\nEvents provided: %s\n", event)
+    event_filters = _maps_events_to_filters(event)
+    if not all_tests and test_ids is None:
+        raise InvalidArgumentValueError(
+            "No tests provided for notification rule. Please provide specific tests using --test-ids option or provide --all-tests flag to select all tests."
+        )
+
+    # If all_tests is provided we ignore test_ids provided in command
+    if all_tests:
+        logger.debug("--all-tests option selected. Ignoring values in --test-ids if provided.")
+        test_ids = None
+
+    if not all_events and not event_filters:
+        raise InvalidArgumentValueError(
+            "No events provided for notification rule. Please provide specific events using --events option or provide --all-events flag to select all events."
+        )
+
+    if all_events:
+        logger.debug("--all-events option selected. Ignoring events provided in --event")
+        event_filters = None
+
+    return TestsNotificationRule(
+        display_name=display_name,
+        action_group_ids=action_groups,
+        event_filters=event_filters,
+        test_ids=test_ids,
+    )
+
+
+def get_notification_rule_update_body(
+    existing_notification_rule,
+    action_groups=None,
+    add_event=None,
+    remove_event=None,
+    display_name=None,
+    test_ids=None,
+    all_tests=False,
+    all_events=False,
+):
+    if all_events and remove_event:
+        raise InvalidArgumentValueError(
+            "Can't provide --remove-event along with --all-event option."
+        )
+
+    if not all_tests and not test_ids:
+        # No change in existing test_ids
+        test_ids = existing_notification_rule.get("testIds", None)
+
+    # If all_tests is provided we ignore test_ids provided in command
+    if all_tests:
+        logger.debug("--all-tests option selected. Ignoring values in --test-ids if provided.")
+        test_ids = None
+
+    event_filters = existing_notification_rule.get("eventFilters", None)
+
+    # Remove events from existing notification rule
+    if remove_event:
+        for event in remove_event:
+            if not event["event-id"] in event_filters:
+                msg = "Invalid --remove-event value. Event-id: %s not present in existing event filters." % event.get(
```

Comment:
Invalid --remove-event value. Event-id: %s does not exist existing event filters.